File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.js
const convertTime = (t) => {
    const [h, m] = t.split(":").map(Number);
    return 60*h + m;
}

function solution(book_time) {
    const sortedBookTime = book_time.map(([s, e]) => [convertTime(s), convertTime(e) + 10]).sort((a, b) => a[0] - b[0]);
    
    let rooms = [];
    for(const [s, e] of sortedBookTime){
        let used = false;
        for(let i = 0 ; i < rooms.length; i++){
            if(rooms[i] <= s){
                rooms[i] = e;
                used = true;
                break;
            }
        }
        
        if(!used) rooms.push(e);
    }
    
    return rooms.length;
}

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.py
def convertTime(t):
    h, m = map(int, t.split(":"))
    return 60*h + m

def solution(book_time):
    sortedBookTime = sorted([[convertTime(s), convertTime(e)+10] for s, e in book_time], key = lambda x: x[0])
    rooms = []
    
    for s, e in sortedBookTime:
        used = False
        for i in range(len(rooms)):
            if rooms[i] <= s:
                rooms[i] = e
                used = True
                break
        if not used:
            rooms.append(e)
        
    return len(rooms)
    